core/jobs/batch_jobs/suggestion_stats_computation_jobs.py

```diff
@@ -97,77 +146,243 @@ def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
         )
 
-        unused_put_result = (
-            user_stats_models
-            | 'Put models into the datastore' >> ndb_io.PutModels()
+        unused_contribution_put_result = (
+            user_contribution_stats_models
+            | 'Put contribution models into the datastore' >> ndb_io.PutModels()
+        )
+        unused_review_put_result = (
+            user_review_stats_models
+            | 'Put review models into the datastore' >> ndb_io.PutModels()
+        )
+        unused_question_contribution_put_result = (
+            user_question_contribution_stats_models
+            | 'Put question contribution models into the datastore' >> ndb_io.PutModels()
+        )
+        unused_question_review_put_result = (
+            user_question_review_stats_models
+            | 'Put question review models into the datastore' >> ndb_io.PutModels()
         )
 
         user_stats_models_job_run_results = (
-            user_stats_models
-            | 'Create job run result' >> (
+            user_contribution_stats_models
+            | 'Create contribution job run result' >> (
+                job_result_transforms.CountObjectsToJobRunResult())
+        )
+        user_review_stats_models_job_run_results = (
+            user_review_stats_models
+            | 'Create review job run result' >> (
+                job_result_transforms.CountObjectsToJobRunResult())
+        )
+        user_question_contribution_stats_models_job_run_results = (
+            user_question_contribution_stats_models
+            | 'Create question contribution job run result' >> (
+                job_result_transforms.CountObjectsToJobRunResult())
+        )
+        user_question_review_stats_models_job_run_results = (
+            user_question_review_stats_models
+            | 'Create question review job run result' >> (
                 job_result_transforms.CountObjectsToJobRunResult())
         )
 
         return (
             (
                 user_stats_error_job_run_results,
-                user_stats_models_job_run_results
+                user_stats_models_job_run_results,
+                user_review_stats_error_job_run_results,
+                user_review_stats_models_job_run_results,
+                user_question_contribution_stats_error_job_run_results,
+                user_question_contribution_stats_models_job_run_results,
+                user_question_review_stats_error_job_run_results,
+                user_question_review_stats_models_job_run_results
             )
             | 'Merge job run results' >> beam.Flatten()
         )
 
     @staticmethod
-    def _generate_stats(
+    def _generate_translation_stats(
         suggestions: Iterable[suggestion_registry.SuggestionTranslateContent],
-        opportunity: Optional[opportunity_domain.ExplorationOpportunitySummary]
+        opportunity: Optional[opportunity_domain.ExplorationOpportunitySummary],
+        translation_contribution_key_required: bool
```

Comment:
Feels a bit awkward to pass in a `bool` that defaults to the contribution version. Can we just pass in a reference to the appropriate model instead? Ditto for the question stats method.

Reply:
But this gives me the following python type error. Could you please let me know why it happens?

`core/jobs/batch_jobs/suggestion_stats_computation_jobs.py:207: error: Argument 3 to "_generate_translation_stats" of "GenerateTranslationContributionStatsJob" has incompatible type "Type[TranslationContributionStatsModel]"; expected "Union[TranslationContributionStatsModel, TranslationReviewStatsModel]"  [arg-type]`

Reply:
I think that just means you need to declare the argument type as `Union[TranslationContributionStatsModel, TranslationReviewStatsModel]`, since it can be either.

Reply:
I have done that as here 
<img width="756" alt="image" src="https://user-images.githubusercontent.com/41409951/228711386-7a599711-5202-40e2-88e8-a9b2380a8a51.png">

And it's been called like this
<img width="814" alt="image" src="https://user-images.githubusercontent.com/41409951/228711476-8c0cb6f7-0256-4a6f-be28-f6dda016ca7d.png">